//
// Copyright 2017 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

// 128-bit bit vector. This protobuf is used for storing and comparing flags.
// The corresponding cc util file implements operators for masking (&) and
// equality testing (==) for this proto.
message BitVector128 {
  // Least-significant 64 bits of the Bitvector.
  optional uint64 low = 1;
  // Most-significant 64 bits of the Bitvector.
  optional uint64 high = 2;
}
